src/Core/Framework/DataAbstractionLayer/Event/BeforeVersionMergeEvent.php

```diff
@@ -0,0 +1,42 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\DataAbstractionLayer\Event;
+
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * @internal
+ *
+ * This event is dispatched during the version merge process and solves specific CMS entities problem.
```

Comment:
I think the comment is a bit problematic :thinking: 

the event itself is not tied to CMS entities and just being an event does not solve any problem :wink: also the last sentence should not be part of this class comment. 

I would put hints about that into the subscriber, where the actual problem is solved.

Reply:
reading this again, all versioned entities have this problem, right? so what is the issue with solving this for all entities? I mean we already have multiple versioned entities :thinking:

Reply:
just saw the other discussion :see_no_evil:

Reply:
https://github.com/shopware/shopware/pull/5843#discussion_r1893675074 :)

Reply:
yeah, but anyway I would rearrange the comments :slightly_smiling_face: